Utility/Utility.Public/Include/Utility/SafeQueue.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <condition_variable>
+#include <deque>
+#include <mutex>
+#include <thread>
+
+namespace Utility
+{
+template <typename T>
+/**
+ * @brief The tamplate SafeQueue class
+ */
+class SafeQueue
+{
+private:
+    std::mutex              _ScopedMutex, _UniqueMutex;
+    std::condition_variable _Block;
+
+    std::deque<T> _RawQueue;
+
+public:
+    /// Default SafeQueue constructor
+    SafeQueue() = default;
+
+    /// SafeQueue copy constructor delete
+    SafeQueue(const SafeQueue<T>& other) = delete;
+
+    /// SafeQueue destructor for clear method
+    ~SafeQueue() { clear(); }
+
+    /// Method for access to front value
+    const T& front()
```

Comment:
The hpp file is the definition, the cpp file is the implementation.

Reply:
I don`t agree. In this case, there is no need to move implementation in cpp file. The implementation is very short, so it may get lost because of its shortness. I don`t see any reasons to transfer implementation in cpp file

Reply:
Here I would move implementation in cpp

Reply:
> I don't agree. In this case, there is no need to move implementation in cpp file. The implementation is very short, so it may get lost because of its shortness. I don't see any reasons to transfer implementation in cpp file

In general, if it's not a class template, you need to take the implementation out of the class. There is a concept of definition and implementation, and you need to separate one from the other. In this instance, everything is fine, but you can also take out the implementation, it will take at most 3 lines of code :)

Reply:
We already decided that the rule about template classes and their implementation exists in our code style guide